hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -384,10 +384,8 @@ public ClosableIterator<T> getClosableIterator() throws IOException {
    * @return An iterator over the records that wraps the engine-specific record in a HoodieRecord.
    */
   public ClosableIterator<HoodieRecord<T>> getClosableHoodieRecordIterator() throws IOException {
-    return new CloseableMappingIterator<>(getClosableIterator(), nextRecord -> {
-      BufferedRecord<T> bufferedRecord = BufferedRecord.forRecordWithContext(nextRecord, readerContext.getSchemaHandler().getRequestedSchema(), readerContext, orderingFieldName, false);
```

Comment:
If we fix the `isDelete` flag here, the new introduced `#constructHoodieRecord` can be eliminated?

Reply:
This is what I recommended to address this: https://github.com/apache/hudi/pull/13445/files/11a4df01b69b973f5fce11fbb574b1edba4ca19c#r2151183064